streaming/base/format/mds/writer.py

```diff
@@ -19,27 +19,42 @@ class MDSWriter(JointWriter):
     """Writes a streaming MDS dataset.
 
     Args:
-        dirname (str): Local dataset directory.
         columns (Dict[str, str]): Sample columns.
+        local: (str, optional): Optional local output dataset directory. If not provided, a random
```

Comment:
Just thinking out loud, should we also have an another optional parameter called src_dir where it contains the raw source files which can be image, text, parquet, etc. ? Then Writer converts the files into a .mds shard files in a `local` directory?

Reply:
we can't make assumptions about where samples are coming from